js/actions/dashboard.js

```diff
@@ -36,11 +45,20 @@ export function setStudentExpanded (studentId, value) {
     if (value) {
       // When user expands student row, check if there's at least one expanded activity.
       // If not, expand the fist one.
-      const expandedActivities = getState().getIn(['dashboard', 'expandedActivities'])
-      if (!expandedActivities.includes(true)) {
-        const firstActivity = getState().getIn(['report', 'activities']).first()
-        dispatch(setActivityExpanded(firstActivity.get('id'), true))
-      }
+      checkActivityExpanded(dispatch, getState)
+    }
+  }
+}
+
+export function setStudentsExpanded (studentIds, value) {
+  return (dispatch, getState) => {
+    dispatch({
```

Comment:
I wasn't sure the best way to dispatch actions to expand/close all the students. I could have just issued `SET_STUDENT_EXPANDED` actions for all students, but I figured that would make the action log unnecessarily confusing. I also considered making a `SET_ALL_STUDENTS_EXPANDED` action, but at that point I need to pass all the student IDs anyway since they're not in the Dashboard's state. This option seemed the most flexible.

Reply:
You're not limited to Dashboard state only. Dashboard reducer is just reducer specific to Dashboard. The app itself uses bunch of common state (report itself). You actually have access to students here:
```javascript
const students = getState().getIn(['report', 'students'])
```
But what you did is okay too.

Reply:
D'oh, right, thanks for the reminder!